app/models/user.rb

```diff
@@ -200,123 +229,72 @@ def can_make_challenges
      challenges. Please wait until they are completed before you create another."]
   end
 
-  def has_clearance?(permission_level)
+  def clearance?(permission_level)
     # for each group (devs, mods, admins) there is a bit in the permission_level
     # indicating whether that group has clearance.
     # a user has clearance if they are a member of any of the groups which has clearance.
     # if all bits are set, then everyone has clearance (including users not in any group);
     # if no bits are set, then no-one has clearance.
     everyone_clearance = permission_level == ~0
-    admin_clearance    = permission_level.anybits?(1 << 0)
-    mod_clearance      = permission_level.anybits?(1 << 1)
-    dev_clearance      = permission_level.anybits?(1 << 2)
+    admin_clearance = permission_level.anybits?(1 << 0)
+    mod_clearance = permission_level.anybits?(1 << 1)
+    dev_clearance = permission_level.anybits?(1 << 2)
 
     everyone_clearance ||
       (is_developer && dev_clearance) ||
       (is_moderator && mod_clearance) ||
-      (is_admin     && admin_clearance)
-  end
-
-  def submission_limit
-    max_dad_level = highest_level
-    return 1 if max_dad_level.zero?
-
-    return max_dad_level if max_dad_level < UNLIMITED_SUBMISSIONS_REQUIREMENT
-
-    -1
+      (is_admin && admin_clearance)
   end
 
   def profile_picture
-    profile_picture = avatar.thumb.url unless avatar.nil?
-
-    if profile_picture.blank?
-      return 'https://s3.us-east-2.amazonaws.com/do-art-daily-public/Default+User+Thumb.png'
-    end
-
-    profile_picture
+    avatar.thumb.url || 'https://s3.us-east-2.amazonaws.com/do-art-daily-public/Default+User+Thumb.png'
   end
 
+  ### Moderation ###
   def approve(reason, moderator)
-    ModeratorLog.create(user_id: moderator.id,
-                        target: self,
-                        action: "#{moderator.username} has approved #{username}.",
-                        reason: reason)
-    update_attribute(:approved, true)
+    create_mod_log(moderator, "#{moderator.username} has approved #{username}.", reason)
```

Comment:
This change to the moderator logs doesn't preserve the target parameter.

Reply:
The target is preserved because of the relation :moderator_logs up a the top:
proof:
![image](https://user-images.githubusercontent.com/36219436/144742999-791d2264-ec48-4000-9ac0-27b2a290c8cb.png)

Reply:
Better proof
![image](https://user-images.githubusercontent.com/36219436/144743035-6edb76f3-80de-4d15-a3e4-dcd048128476.png)